### PR TITLE
added custom zap logger config

### LINKED ISSUE #1152: Timestamps in nexd logs prevent log summarization
Here is a sample `nexd` log message:

```
{"level":"info","ts":1686166202.684511,"caller":"nexodus/wg.go:233","msg":"Removed peer with key 5l0AHeRRGW1xeTeehW3Ab3P2cTfarCBTBZwGZV/9XSM="}
```

The `ts` field in the log message prevents a logging daemon from summarizing repeated messages. We should leave timestamps to logging daemons to provide.

### PR DESCRIPTION
**Pull request by akoserwal**
Closes #1152

Added a custom zap custom configuration to remove `ts` from the logs

### CODE REVIEW

**Comment by akoserwal** [comment]
@russellb

File: cmd/nexd/main.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package main
 import (
 	"context"
 	"fmt"
+	"go.uber.org/zap/zapcore"
```
Comment (russellb) [inline cmd/nexd/main.go:6]: Can you move this down to be with the import of `zap` ?

File: cmd/nexd/main.go [hunk 2]
```diff
@@ -148,20 +149,39 @@ func nexdRun(cCtx *cli.Context, logger *zap.Logger, logLevel *zap.AtomicLevel, m
...
 	if debug != "" {
-		logCfg := zap.NewDevelopmentConfig()
-		logLevel = &logCfg.Level
+		logCfg := NewCustomConfig()
 		logger, err = logCfg.Build()
 		logger.Info("Debug logging enabled")
 	} else {
-		logCfg := zap.NewProductionConfig()
-		logLevel = &logCfg.Level
+		logCfg := NewCustomConfig()
```
Comment (russellb) [inline cmd/nexd/main.go:184]: How about still using zap's default production vs. debug configurations, but just setting `logCfg.EncoderConfig.TimeKey = ""` ?

**Review by russellb — CHANGES_REQUESTED** [review]
Thanks for looking at this! I have a couple of small suggestions.

File: cmd/nexd/main.go (hunk 2 above)
Comment (akoserwal) [reply cmd/nexd/main.go]: Yes, it makes sense. I update the PR.

**Review by russellb — APPROVED** [review]
Thank you!

**Pull request merged**